homeassistant/components/nightscout/config_flow.py

```diff
@@ -40,19 +41,26 @@ async def async_step_user(self, user_input=None):
         errors = {}
         if user_input is not None:
             try:
-                info = await _validate_input(user_input)
-            except CannotConnect:
-                errors["base"] = "cannot_connect"
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
-            else:
-                return self.async_create_entry(title=info["title"], data=user_input)
+                return await self._try_create_entry(user_input)
```

Comment:
I don't really like that we're wrapping so much code in the try... except that doesn't need to catch the exception. I'd suggest to remove the `_try_create_entry` method and inline that logic in the user step, and only wrap the `_validate_input` call in the try... except block. Use an `else:` block if needed.